topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py

```diff
@@ -27,6 +27,8 @@ def is_applicable(self, asset: Asset) -> bool:
     def load_metadata(self, asset: Asset) -> None:
 
         fs = get_fs(asset.source_path)
+        # FIXME: Should we download the file first as we could need it to do the coggification later?
```

Comment:
Reading a tiff header should be almost instant, It is generally one read from the head of the file.

Is rasterio reading a lot more than it should? Is there a way to force rasterio to only read the tiff headers?

Reply:
I haven't investigate, but I can say it's quiet long!

Reply:
What is "quite long"?

Reply:
20-30 seconds

Reply:
for how many files?

Reply:
Per file!

Reply:
```json
{"loader": "metadata.loader.imagery.tiff", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054d30>", "duration": 28361.980224609375, "level": 20, "time": 1638101651317, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.historic", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e1feb0a0>", "duration": 1.566162109375, "level": 20, "time": 1638101651319, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.tiff", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e1feb0a0>", "duration": 17520.66162109375, "level": 20, "time": 1638101668840, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.historic", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054730>", "duration": 4.18115234375, "level": 20, "time": 1638101668845, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.tiff", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054730>", "duration": 22965.641845703125, "level": 20, "time": 1638101691811, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.historic", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054070>", "duration": 2.7060546875, "level": 20, "time": 1638101691815, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.tiff", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054070>", "duration": 24532.73388671875, "level": 20, "time": 1638101716348, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.historic", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054be0>", "duration": 1.12060546875, "level": 20, "time": 1638101716349, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.tiff", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054be0>", "duration": 25956.140625, "level": 20, "time": 1638101742305, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.historic", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054d90>", "duration": 1.721923828125, "level": 20, "time": 1638101742307, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.tiff", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054d90>", "duration": 21477.30859375, "level": 20, "time": 1638101763785, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.historic", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054280>", "duration": 4.956787109375, "level": 20, "time": 1638101763791, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.tiff", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054280>", "duration": 25951.318115234375, "level": 20, "time": 1638101789743, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
```
Where `duration` is in `ms`

Reply:
That is really really slow :( Basemaps's Reads EPSG/file size from a 1000 files in few seconds... wonder why this is 1000x slower